components/InteractiveElements/textInputs/CustomTextInput.tsx

```diff
@@ -0,0 +1,130 @@
+import GraphemeSplitter from "grapheme-splitter";
+import { useEffect, useState } from "react";
+import { View, TextInput } from "react-native";
+import { IconButton, Text } from "react-native-paper";
+import { useSelector } from "react-redux";
+
+import { RootState } from "../../../types";
+
+export default function CustomTextInput({
+  text,
+  setText,
+  textLimit = 70,
```

Comment:
Here it looks like textLimit is still 70. We could create 2 different limits in the constants file for name and message limits, and pass them in conditionally.

Reply:
> For me I'm getting a limit of 70 characters (not 15 as you describe). Screenshots attached. Could you take a look? I think the letter limit passed in might still be 70 from what I can see in the code.
> 
> Either way I'm wondering if 15 is too short and we should go with 20? although I'd be fine with 15 too if this makes things look bad for everyone, it's prob enough to cover the majority of first names, and your component tells the user what the char limit is. ![image](https://user-images.githubusercontent.com/59271264/173716069-a9caaff6-6e8d-4b05-b6e0-368142f425eb.png)
> 
> ![image](https://user-images.githubusercontent.com/59271264/173716082-45e57bef-c5d0-432f-893e-202c57437425.png)

after our meeting on 5/24, I pushed a commit so that both Names and Messages share a 70-character limit

Reply:
Also, yes, I will make an update to move these values to the constants folder!

Reply:
Oh right sorry, I forgot we decided to do 70 for name as well so I referenced the PR description instead. 

Right now as you can see on my iPhone screenshot, if I do the 70 max, it goes off the screen and the "created by" overlaps the download button. Does it look like that on yours? 

I feel like we should either fix the puzzle solve page so that it looks ok with the max length name, or lower the max length name limit so that it still looks ok on the puzzle solve page (in which case I guess some logic might be required to differentiate which limit we pass into the component). Thoughts?

Reply:
good catch! I'm inclined to fix the puzzle solve page to look ok with the max length name and will work on that this weekend.